cmd/find-main.go

```diff
@@ -0,0 +1,191 @@
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find object names matching wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Match directory names matching wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Match directory and object name with PCRE regex pattern",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Print in custom format to STDOUT (see FORMAT)",
+		},
+		cli.StringFlag{
+			Name:  "exec",
+			Usage: "Spawn an external process for each matching object (see FORMAT)",
+		},
+		cli.StringFlag{
+			Name:  "ignore",
+			Usage: "Exclude objects matching the wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "newer",
+			Usage: "Match all objects newer than specified time in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "older",
+			Usage: "Match all objects older than specified time in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "larger",
+			Usage: "Match all objects larger than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "smaller",
+			Usage: "Match all objects smaller than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "maxdepth",
+			Usage: "Limit directory navigation to specified depth",
+		},
+		cli.BoolFlag{
+			Name:  "watch",
+			Usage: "Monitor specified location for newly created objects",
+		},
+		cli.BoolFlag{
+			Name:  "or",
+			Usage: "Changes the matching criteria from an \"and\" to an \"or\"",
+		},
+	}
+)
+
+var findCmd = cli.Command{
+	Name:   "find",
+	Usage:  "Finds files which match the given set of parameters.",
+	Action: mainFind,
+	Flags:  append(findFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} PATH FLAG EXPRESSION [FLAG]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+   1. Find all files named foo from all buckets.
+   	$ {{.HelpName}} s3 --name "file"
+   
+   2. Find all text files from mybucket.
+   	$ {{.HelpName}} s3/mybucket --name "*.txt"
+   
+   3. Print only the object names without the directory component under this bucket.
+   	$ {{.HelpName}} s3/bucket --name "*" -print {base}
+   
+   4. Copy all jpg files from AWS S3 photos bucket to minio play test bucket.
+   	$ {{.HelpName}} s3/photos --name "*.jpg" --exec "mc cp {} play/test"
+
+   5. Find all jpg images from any folder prefixed with album.
+   	$ {{.HelpName}} s3/photos --name "*.jpg" --path "*/album*/*"
+   
+   6. Find all jpgs, pngs, and gifs using regex
+   	$ {{.HelpName}} s3/photos --regex "(?i)\.(jpg|png|gif)$"
+
+   7. Mirror all photos from s3 bucket *coninuously* from the s3 bucket to minio play test bucket.
+   	$ {{.HelpName}} s3/buck --name "*foo" --watch --exec "mc cp {} play/test"	
+
+   8. Generate self expiring urls (7 days), for all objects between 64 MB, and 1 GB in size. 
+   	$ {{.HelpName}} s3 --larger 64MB --smaller 1GB --print {url}
```

Comment:
AB chose this one, I think because the original ideas "purl" or "presigned url" were thought to be more confusing and that url was very concise and easy to understand/.